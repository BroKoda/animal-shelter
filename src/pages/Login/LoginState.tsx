import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'

export interface LoginState {
  userToLogin: LoginDetails
  loggedInUser?: any
  error?: string
  status: LoadingStatus
}

export interface LoginDetails {
  email?: string
  password?: string
}