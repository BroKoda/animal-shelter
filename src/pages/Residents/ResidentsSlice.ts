import { Animal, Resident, ResidentsState } from './ResidentsState'
import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'
import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { RootState } from '../../app/store'
import { addDoc, collection, doc, getDoc, getDocs } from 'firebase/firestore'
import { getDownloadURL, ref, uploadBytes } from 'firebase/storage'
import { db, storage } from '../../firebase'

interface ImageToUpload {
  image: File,
  name: string
}

const initialState: ResidentsState = {
  residents: [],
  residentToAdd: {},
  residentDetails: {},
  fetchResidentsStatus: LoadingStatus.initial,
  addNewResidentStatus: LoadingStatus.initial,
  addNewResidentImageStatus: LoadingStatus.initial,
  fetchResidentImageStatus: LoadingStatus.initial,
  fetchResidentDetailsStatus: LoadingStatus.initial
}

export const fetchResidents = createAsyncThunk('fetchResidents', async () => {
  try {
    const data = await getDocs(collection(db, 'animals'))
    const list: Resident[] = []
    data.forEach((doc) => {
      list.push({id: doc.id, ...doc.data()})
    })
    return list
  } catch (e) {
    console.log(e)
    return undefined
  }
})

export const addNewResident = createAsyncThunk('addNewResident', async (animal: Animal) => {
  if (animal != null) {
    try {
      await addDoc(collection(db, 'animals'), {
        animal: animal
      })
    } catch (e) {
      console.log(e)
    }
  }
})

export const addNewResidentImage = createAsyncThunk('addNewResidentIamage', async ({image, name}: ImageToUpload) => {
  if (image != null) {
    try {
      const imageRef = ref(storage, `resident-images/${name}`)
      uploadBytes(imageRef, image).then(() => {
        console.log('Image uploaded')
      })
      return name
    } catch (e) {
      console.log(e)
    }
  }
})

export const fetchResidentDetails = createAsyncThunk('fetchResidentDetails', async (id: string) => {
  try {
    const data = await getDoc(doc(db, 'animals', id))
    return data.data()
  } catch (e) {
    console.log(e)
    return undefined
  }
})

export const fetchResidentImage = createAsyncThunk('fetchResidentImage', async (image: string) => {
  try {
    const pathRef = ref(storage, `resident-images/${image}`)
    return await getDownloadURL(pathRef)
  }
  catch (e) {
    console.log(e)
    return undefined
  }
})

const residentsSlice = createSlice({
  name: 'residents',
  initialState,
  reducers: {
    resetResidents: () => initialState,
    setName: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.name = action.payload
    },
    setType: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.type = action.payload
    },
    setColor: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.color = action.payload
    },
    setSize: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.size = action.payload
    },
    setBirtDate: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.birthDate = action.payload
    },
    setArrivalDate: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.arrivalDate = action.payload
    },
    setDescription: (state: ResidentsState, action: PayloadAction<string>) => {
      state.residentToAdd.description = action.payload
    },
  },
  extraReducers: builder => {
    builder
      .addCase(fetchResidents.pending, (state) => {
        state.fetchResidentsStatus = LoadingStatus.loading
      })
      .addCase(fetchResidents.fulfilled, (state, action) => {
        state.fetchResidentsStatus = LoadingStatus.complete
        state.residents = action.payload
      })
      .addCase(fetchResidents.rejected, (state) => {
        state.fetchResidentsStatus = LoadingStatus.error
      })
      .addCase(addNewResident.pending, (state) => {
        state.addNewResidentStatus = LoadingStatus.loading
      })
      .addCase(addNewResident.fulfilled, (state) => {
        state.addNewResidentStatus = LoadingStatus.complete
      })
      .addCase(addNewResident.rejected, (state) => {
        state.addNewResidentStatus = LoadingStatus.error
      })
      .addCase(fetchResidentDetails.pending, (state) => {
        state.fetchResidentDetailsStatus = LoadingStatus.loading
      })
      .addCase(fetchResidentDetails.fulfilled, (state, action) => {
        state.fetchResidentDetailsStatus = LoadingStatus.complete
        state.residentDetails = action.payload
      })
      .addCase(fetchResidentDetails.rejected, (state) => {
        state.fetchResidentDetailsStatus = LoadingStatus.error
      })
      .addCase(addNewResidentImage.pending, (state) => {
        state.addNewResidentImageStatus = LoadingStatus.loading
      })
      .addCase(addNewResidentImage.fulfilled, (state, action) => {
        state.addNewResidentImageStatus = LoadingStatus.complete
        state.residentToAdd.image = action.payload
      })
      .addCase(addNewResidentImage.rejected, (state) => {
        state.addNewResidentImageStatus = LoadingStatus.error
      })
      .addCase(fetchResidentImage.pending, (state) => {
        state.fetchResidentImageStatus = LoadingStatus.loading
      })
      .addCase(fetchResidentImage.fulfilled, (state, action) => {
        state.fetchResidentImageStatus = LoadingStatus.complete
        state.residentDetails?.animal != null && (state.residentDetails.animal.imageUrl = action.payload)
      })
      .addCase(fetchResidentImage.rejected, (state) => {
        state.fetchResidentImageStatus = LoadingStatus.error
      })
  }
})

export const {
  resetResidents,
  setName,
  setType,
  setColor,
  setSize,
  setBirtDate,
  setArrivalDate,
  setDescription
} = residentsSlice.actions
export const selectResidents = (state: RootState) => state.residents
export default residentsSlice.reducer