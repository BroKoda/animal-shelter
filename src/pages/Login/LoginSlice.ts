import { LoginDetails, LoginState, User } from './LoginState'
import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'
import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { RootState } from '../../app/store'
import { getAuth, signInWithEmailAndPassword, signOut } from 'firebase/auth'
import { collection, getDocs, query, where } from 'firebase/firestore'
import { db } from '../../firebase'

const initialState: LoginState = {
  userToLogin: {},
  status: LoadingStatus.initial,
  userStatus: LoadingStatus.initial,
  signOutUserStatus: LoadingStatus.initial
}

export const logInUser = createAsyncThunk('logInUser', async ({email, password}: LoginDetails) => {
  if (email != null && password != null) {
    await signInWithEmailAndPassword(getAuth(), email, password)
    return true
  }
})

export const getUserDetails = createAsyncThunk('getUserDetails', async (email: string | null) => {
  if (email != null) {
    const userData = await getDocs(query(collection(db, 'users'), where('email', '==', email)))
    return userData.docs[0].data() as User
  }
})

export const signOutUser = createAsyncThunk('signOutUser', async () => {
  const auth = getAuth()
  await signOut(auth)
  return true
})

const loginSlice = createSlice({
  name: 'login',
  initialState,
  reducers: {
    resetLogin: () => initialState,
    setLoginEmail: (state: LoginState, action: PayloadAction<string>) => {
      state.userToLogin.email = action.payload
    },
    setLoginPassword: (state: LoginState, action: PayloadAction<string>) => {
      state.userToLogin.password = action.payload
    },
  },
  extraReducers: builder =>  {
    builder
      .addCase(logInUser.pending, (state) => {
        state.status = LoadingStatus.loading
      })
      .addCase(logInUser.fulfilled, (state) => {
        state.status = LoadingStatus.complete
      })
      .addCase(logInUser.rejected, (state) => {
        state.status = LoadingStatus.error
        state.error = 'Nem megfelelő email cím vagy jelszó!'
      })
      .addCase(getUserDetails.pending, (state) => {
        state.userStatus = LoadingStatus.loading
      })
      .addCase(getUserDetails.fulfilled, (state, action) => {
        state.userStatus = LoadingStatus.complete
        state.user = action.payload
      })
      .addCase(getUserDetails.rejected, (state) => {
        state.userStatus = LoadingStatus.error
      })
      .addCase(signOutUser.pending, (state) => {
        state.signOutUserStatus = LoadingStatus.loading
      })
      .addCase(signOutUser.fulfilled, (state) => {
        state.signOutUserStatus = LoadingStatus.complete
      })
      .addCase(signOutUser.rejected, (state) => {
        state.signOutUserStatus = LoadingStatus.error
      })
  }
})

export const { resetLogin, setLoginEmail, setLoginPassword } = loginSlice.actions
export const selectLogin = (state: RootState) => state.login
export default loginSlice.reducer
