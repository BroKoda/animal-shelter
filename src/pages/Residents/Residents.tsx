import React, { useEffect } from 'react'
import '../../assets/pages/Residents.scss'
import BaseLayout from '../../layout/BaseLayout'
import { useAppDispatch, useAppSelector } from '../../app/hooks'
import { fetchResidents } from './ResidentsSlice'
import AnimalCard from '../../components/AnimalCard/AnimalCard'
import { Resident } from './ResidentsState'

const Residents = (): JSX.Element => {
  const dispatch = useAppDispatch()
  const residents = useAppSelector((state) => state.residents.residents)

  useEffect(() => {
    void dispatch(fetchResidents())
  }, [dispatch])

  return (
    <BaseLayout>
      <div className='container residents-container'>
        <div className='row'>
          <div className='col-12'>
            <h1>Lakóink</h1>
          </div>
        </div>
        <div className='row'>
          {residents.map((resident: Resident, index: number) => {
            if (resident.animal != null) {
              return (
                <div key={index} className='col-4'>
                  <AnimalCard key={index} resident={resident.animal}/>
                </div>
              )
            } else {
              return "Jelenleg nincs lakó a menhelyen!"
            }
          })}
        </div>
      </div>
    </BaseLayout>
  )
}

export default Residents