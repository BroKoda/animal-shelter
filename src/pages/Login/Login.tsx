import React, { ChangeEvent, useCallback, useEffect } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import '../../assets/pages/Login.scss'
import { useAppDispatch, useAppSelector } from '../../app/hooks'
import { logInUser, resetLogin, setLoginEmail, setLoginPassword } from './LoginSlice'
import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'

const Login = (): JSX.Element => {
  const navigate = useNavigate()
  const dispatch = useAppDispatch()
  const { status, error } = useAppSelector((state) => state.login)
  const { email, password } = useAppSelector((state) => state.login.userToLogin)

  useEffect(() => {
    if (status === LoadingStatus.complete) {
      dispatch(resetLogin())
      navigate('/')
    }
  }, [dispatch, status])

  const handleLogin = useCallback(() => {
    void dispatch(logInUser({email, password}))
  }, [dispatch, email, password])

  const handleSetLoginEmail = useCallback((event: ChangeEvent<HTMLInputElement>) => {
    dispatch(setLoginEmail(event.target.value))
  }, [dispatch, email])

  const handleSetLoginPassword = useCallback((event: ChangeEvent<HTMLInputElement>) => {
    dispatch(setLoginPassword(event.target.value))
  }, [dispatch, password])

  return (
    <div className="container login-container">
      <div className="row justify-content-center h-100">
        <div className="col-12 col-md-8 col-lg-6 col-xl-5 align-self-center">
          <div className="container login-box py-4 px-4">
            <div className="row">
              <div className="col-12">
                <h2>Belépés</h2>
                <p className='login-error'>{error}</p>
                <form className='my-4'>
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-at"></i>
                    </div>
                    <input
                      type="email"
                      className="form-control"
                      placeholder='Email'
                      id="loginEmail"
                      value={email ?? ''}
                      onChange={handleSetLoginEmail}
                    />
                  </div>
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-lock"></i>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      placeholder='Jelszó'
                      id="loginPassword"
                      value={password ?? ''}
                      onChange={handleSetLoginPassword}
                    />
                  </div>
                </form>
              </div>
            </div>
            <div className='row'>
              <div className='col-12 mb-3 text-end redirect-to-registration'>
                <Link to={'/regisztracio'}>
                  <span>Nincs még felhasználód? Regisztrálj!</span>
                </Link>
              </div>
            </div>
            <div className='row'>
              <div className='col-6'>
                <Link to={'/'}>
                  <button className='button secondary-button'>
                    Vissza
                  </button>
                </Link>
              </div>
              <div className='col-6'>
                <button className='button call-to-action-button' onClick={handleLogin}>
                  Belépés
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Login
