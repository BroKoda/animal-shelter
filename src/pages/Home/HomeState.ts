import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'
import { News } from '../News/NewsState'
import { Resident } from '../Residents/ResidentsState'

export interface HomeState {
  news?: News[]
  residents?: Resident[]
  fetchNewsStatus: LoadingStatus
  fetchResidentsStatus: LoadingStatus
}