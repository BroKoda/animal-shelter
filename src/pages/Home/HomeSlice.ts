import { HomeState } from './HomeState'
import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
import { RootState } from '../../app/store'
import { collection, getDocs, query, limit, orderBy } from 'firebase/firestore'
import { db } from '../../firebase'
import { Resident } from '../Residents/ResidentsState'
import { News } from '../News/NewsState'

const initialState: HomeState = {
  news: [],
  residents: [],
  fetchNewsStatus: LoadingStatus.initial,
  fetchResidentsStatus: LoadingStatus.initial
}

export const fetchHomeResidents = createAsyncThunk('fetchHomeResidents', async () => {
  try {
    const data = await getDocs(query(collection(db, 'animals'), limit(3)))
    const list: Resident[] = []
    data.forEach((doc) => {
      list.push({id: doc.id, ...doc.data()})
    })
    return list
  } catch (e) {
    console.log(e)
    return undefined
  }
})

export const fetchHomeNews = createAsyncThunk('fetchHomeNews', async () => {
  try {
    const data = await getDocs(query(collection(db, 'news'), limit(3)))
    const list: News[] = []
    data.forEach((doc) => {
      list.push({id: doc.id, ...doc.data()})
    })
    return list
  } catch (e) {
    console.log(e)
    return undefined
  }
})

const homeSlice = createSlice({
  name: 'home',
  initialState,
  reducers: {},
  extraReducers: builder => {
    builder
      .addCase(fetchHomeResidents.pending, (state) => {
        state.fetchResidentsStatus = LoadingStatus.loading
      })
      .addCase(fetchHomeResidents.fulfilled, (state, action) => {
        state.fetchResidentsStatus = LoadingStatus.complete
        state.residents = action.payload
      })
      .addCase(fetchHomeResidents.rejected, (state) => {
        state.fetchResidentsStatus = LoadingStatus.error
      })
      .addCase(fetchHomeNews.pending, (state) => {
        state.fetchNewsStatus = LoadingStatus.loading
      })
      .addCase(fetchHomeNews.fulfilled, (state, action) => {
        state.fetchNewsStatus = LoadingStatus.complete
        state.news = action.payload
      })
      .addCase(fetchHomeNews.rejected, (state) => {
        state.fetchNewsStatus = LoadingStatus.error
      })
  }
})

export const selectHome = (state: RootState) => state.home
export default homeSlice.reducer