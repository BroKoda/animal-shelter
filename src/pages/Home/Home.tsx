import React, { useEffect } from 'react'
import BaseLayout from '../../layout/BaseLayout'
import { Resident } from '../Residents/ResidentsState'
import AnimalCard from '../../components/AnimalCard/AnimalCard'
import { useAppDispatch, useAppSelector } from '../../app/hooks'
import { fetchHomeNews, fetchHomeResidents } from './HomeSlice'
import { News } from '../News/NewsState'
import NewsCard from '../../components/NewsCard/NewsCard'

const Home = (): JSX.Element => {
  const dispatch = useAppDispatch()
  const { residents, news } = useAppSelector((state) => state.home)

  useEffect(() => {
    dispatch(fetchHomeResidents())
  }, [dispatch])

  useEffect(() => {
    dispatch(fetchHomeNews())
  }, [dispatch])

  return (
    <BaseLayout>
      <div className='container'>
        <div className='row'>
          <div className='col-12 mb-4'>
            <h1>Mancs állatvédő egyesület</h1>
          </div>
        </div>
        <div className='row'>
          <div className='col-12'>
            <p>Már mintegy 20 éve ad testi-lelki ápolást, otthont a rászoruló állatoknak, illetve segít nekik
              gazdához és boldog élethez jutni. Motivációnk a feltétlen, önzetlen állatszeretet. Célunk, hogy az
              emberekhez és állatokhoz egyaránt méltó, tiszta, modern, hatékony ellátást és környezetet tudjunk
              biztosítani a gondozottjaink és a hozzánk ellátogató családok részére.</p>
            <p className='mt-3'>Működésünk a Magyarkanizsai Önkormányzat, a felajánlott személyi jövedelemadó 1%-ok és
              magánszemélyek adományai segítségével, illetve a menhely védjegyévé vált önkéntes csapat által
              biztosított.</p>
          </div>
        </div>
        <div className='row'>
          <div className='col-12 mt-5'>
            <h3>Aktuális híreink</h3>
          </div>
        </div>
        <div className='row'>
          <div className='col-12 mb-3 mt-2'>
            <p>Itt találod a legfrissebb híreinket. Kattints
              ide
              az összes hír megtekintéséhez!</p>
          </div>
        </div>
        <div className='row'>
          {news != null && news.map((singleNews: News, index: number) => {
            if (singleNews.newsDetails != null) {
              return (
                <div key={index} className="col-4">
                  <NewsCard key={index} news={singleNews.newsDetails} id={singleNews.id}/>
                </div>
              )
            } else {
              return 'Jelenleg nincs elérhető hír!'
            }
          })}
        </div>
        <div className='row'>
          <div className='col-12 mt-5'>
            <h3>Fogadj örökbe!</h3>
          </div>
        </div>
        <div className='row'>
          <div className='col-12 mb-3 mt-2'>
            <p>Itt találod a legújabb lakóinkat. Kattints
              ide
              az összes lakónk megtekintéséhez!</p>
          </div>
        </div>
        <div className='row'>
          {residents != null && residents.map((resident: Resident, index: number) => {
            if (resident.animal != null) {
              return (
                <div key={index} className='col-4'>
                  <AnimalCard key={index} resident={resident.animal} id={resident.id} />
                </div>
              )
            } else {
              return "Jelenleg nincs lakó a menhelyen!"
            }
          })}
        </div>
      </div>
    </BaseLayout>
  )
}

export default Home