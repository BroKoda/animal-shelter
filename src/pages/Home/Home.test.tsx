import { renderWithProviders } from '../../utils/test-utils'
import { BrowserRouter } from 'react-router-dom'
import Home from './Home'
import { LoginState } from '../Login/LoginState'
import LoginSlice, { selectLogin, signOutUser } from '../Login/LoginSlice'
import { RootState, store } from '../../app/store'
import { LoadingStatus } from '../../components/LoadingStatus/LoadingStatus'
import { HomeState } from './HomeState'
import HomeSlice, { fetchHomeNews, fetchHomeResidents, selectHome } from './HomeSlice'
import { fetchResidents } from '../Residents/ResidentsSlice'

describe('Home page tests', () => {
  const appState = store.getState()
  const initialState: HomeState = {
    news: [],
    residents: [],
    fetchNewsStatus: LoadingStatus.initial,
    fetchResidentsStatus: LoadingStatus.initial
  }
  it('Should render home page', async () => {
    const { container } = renderWithProviders(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('Should set loading state when fetching news for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeNews.pending)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchNewsStatus).toEqual(LoadingStatus.loading)
    done()
  })

  it('Should set loading state when fetching news for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeNews.fulfilled)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchNewsStatus).toEqual(LoadingStatus.complete)
    done()
  })

  it('Should set loading state when fetching news for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeNews.rejected)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchNewsStatus).toEqual(LoadingStatus.error)
    done()
  })

  it('Should set loading state when fetching residents for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeResidents.pending)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchResidentsStatus).toEqual(LoadingStatus.loading)
    done()
  })

  it('Should set loading state when fetching residents for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeResidents.fulfilled)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchResidentsStatus).toEqual(LoadingStatus.complete)
    done()
  })

  it('Should set loading state when fetching residents for home page', (done) => {
    const nextState: HomeState = HomeSlice(initialState, fetchHomeResidents.rejected)
    const rootState: RootState = {... appState, home: nextState}

    expect(selectHome(rootState).fetchResidentsStatus).toEqual(LoadingStatus.error)
    done()
  })
})