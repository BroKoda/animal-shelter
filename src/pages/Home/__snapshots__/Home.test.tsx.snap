// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests Should render home page 1`] = `
<div>
  <div
    class="container-fluid header-container"
  >
    <div
      class="row h-100"
    >
      <div
        class="col-12 align-self-center"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-4 d-md-none align-self-center"
            >
              <i
                class="fa-solid fa-bars js-navigation-toggle"
              />
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-center text-md-start"
            >
              <a
                href="/"
              >
                <img
                  alt="Mancs logo"
                  class="d-none d-md-block"
                  src="/mancs.png"
                />
                <img
                  alt="Mancs logo"
                  class="d-block d-md-none mx-auto"
                  src="/mancs_logo_only.png"
                />
              </a>
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-end"
            >
              <a
                class="d-none d-md-inline-block"
                href="/regisztracio"
              >
                <button
                  class="button secondary-button me-2"
                >
                  Regisztráció
                </button>
              </a>
              <a
                href="/bejelentkezes"
              >
                <button
                  class="button call-to-action-button login-button"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                    Bejelentkezés
                  </span>
                  <span
                    class="d-inline d-md-none"
                  >
                    <i
                      class="fa-solid fa-user"
                    />
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container body-container mt-3 mb-5"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 col-lg-3 navigation-container d-none d-md-block pe-md-0 pe-xl-2"
      >
        <div
          class="container menu-container px-0"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <a
                href="/"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-house"
                    />
                  </div>
                  <span>
                    Főoldal
                  </span>
                </div>
              </a>
              <a
                href="/lakok"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-paw"
                    />
                  </div>
                  <span>
                    Lakóink
                  </span>
                </div>
              </a>
              <a
                href="/hirek"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-newspaper"
                    />
                  </div>
                  <span>
                    Hírek
                  </span>
                </div>
              </a>
              <a
                href="/csatlakozz"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-hand-holding-heart"
                    />
                  </div>
                  <span>
                    Csatlakozz!
                  </span>
                </div>
              </a>
              <a
                href="/kapcsolat"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-location-dot"
                    />
                  </div>
                  <span>
                    Kapcsolat
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-12 col-md-8 col-lg-9"
      >
        <div
          class="container body-container px-0"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-12 col-sm-10 col-md-12"
            >
              <div
                class="progress-indicator-container"
              >
                <span
                  class="progress-indicator"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid footer-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container py-5"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-5 col-lg-4 mb-4 mb-md-0"
            >
              <h5>
                Mancs állatvédő egyesület
              </h5>
              <p
                class="footer-description"
              >
                Tevékenységeink és céljaink magába foglalják az egyhagyott, kidobott, megtalált kölyök és felnőtt állatok mentését, gondozását, nevelését és legfőképp gazdához jutását. Hisszuk, hogy minden állatnak van megfelelő gazdája, mi segítünk megtalálni!
              </p>
            </div>
            <div
              class="col-6 col-md-4 offset-lg-1 useful-links"
            >
              <h5>
                Hasznos linkek
              </h5>
              <a
                href="/"
              >
                Főoldal
              </a>
              <a
                href="/lakok"
              >
                Lakóink
              </a>
              <a
                href="/hirek"
              >
                Hírek
              </a>
              <a
                href="/csatlakozz"
              >
                Csatlakozz!
              </a>
              <a
                href="/kapcsolat"
              >
                Kapcsolat
              </a>
            </div>
            <div
              class="col-6 col-md-3 social-links-container"
            >
              <h5>
                Kapcsolat
              </h5>
              <p>
                Beogradska 3-5
                <br />
                24420 Magyarkanizsa
                <br />
                Szerbia
              </p>
              <h5
                class="mt-4"
              >
                Kövess minket
              </h5>
              <a
                href="/facebook"
                style="text-decoration: none;"
              >
                <i
                  class="fa-brands fa-facebook"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid copyright-strip-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-12 text-center"
            >
              <p>
                © 2023. Minden jog fenntartva. Mancs Állatvédő Egyesület Magyarkanizsa
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
